### ISSUE TITLE
Skipped macOS application code signing  reason=cannot find valid "Mac Developer" identity or custom non-Apple code signing certificate

### ISSUE THREAD

**Issue by johannesjo** [electron-userland/electron-builder#4295]
* **Version**: 21.2.0

* **Target**: Mac OS

Similar to #2513 but none of the solutions worked and at least to me it looks like that all certificates are in place...

```
electron-builder
  • electron-builder  version=21.2.0 os=18.7.0
  • loaded configuration  file=/Users/johannesme/electron-builder.yaml
  • writing effective config  file=app-builds/builder-effective-config.yaml
  • packaging       platform=darwin arch=x64 electron=6.0.7 appOutDir=app-builds/mac
  • packaging       platform=mas arch=x64 electron=6.0.7 appOutDir=app-builds/mas
  • skipped macOS application code signing  reason=cannot find valid "Mac Developer" identity or custom non-Apple code signing certificate, see https://electron.build/code-signing allIdentities=
  1) 33C5A99173A572E577DAF879FF82DDAE9DE2EFAE "iPhone Developer: xxx@xxx.de (XXXXXXXXX)"
  2) 3ACA6E5506DA323306928AB0CFFAEB59C8568E86 "member: 04B3A415-CBFA-4D92-82D8-12619B71E8B2 28326F34-2G0B-403A-A2BF-B390AB7466DC" (CSSMERR_TP_NOT_TRUSTED)
  3) 3ACE07ED89D1ASDE182891B125D5D3E2B9D46B6F "Developer ID Installer: Johannes Me (XXXXXXXXX)"
  4) 33C4C262D5818C82FE57D9D8757327380CC3BB90 "Developer ID Application: Johannes Me (XXXXXXXXX)"
  5) 3DC0CFA50D30912D2A2C61A73AAE31052FB8533D "3rd Party Mac Developer Application: Johannes Me (XXXXXXXXX)"
  6) 34C786EC92AE42D97071E296CCB1A36D3DC7E645 "3rd Party Mac Developer Installer: Johannes Me (XXXXXXXXX)"
  7) 39C9AE8406A568E8B8CB19908D3C7BFE087F0838 "Apple Distribution: Johannes Me (XXXXXXXXX)"
     7 identities found

  Valid identities only
  1) 33C5A99173A572E577DAF879FF82DDAE9DE2EFAE "iPhone Developer: m.schmidt@wind-energie.de (EAT66RHC99)"
  2) 3ACE07ED89D1AB5E182891B125D5D3E2B9D46B6F "Developer ID Installer: Johannes Me (XXXXXXXXX)"
  3) 33C4C262D5818C82FE57D9D8757327380CC3BB90 "Developer ID Application: Johannes Me (XXXXXXXXX)"
  4) 3DC0CFA50D30912D2A2C61A73AAE31052FB8533D "3rd Party Mac Developer Application: Johannes Me (XXXXXXXXX)"
  5) 34C786EC92AE42D97071E296CCB1A36D3DC7E645 "3rd Party Mac Developer Installer: Johannes Me (XXXXXXXXX)"
  6) 39C9AE8406A568E8B8CB19908D3C7BFE087F0838 "Apple Distribution: Johannes Me (XXXXXXXXX)"
     6 valid identities found
  • signing         file=app-builds/mas/superProductivity.app identityName=3rd Party Mac Developer Application: Johannes Me (XXXXXXXXX) identityHash=ED80CFA50D30912D2A2C61A73AAE31052FB8533D provisioningProfile=embedded.provisionprofile
  • skipped macOS application code signing  reason=cannot find valid "Mac Developer" identity or custom non-Apple code signing certificate, see https://electron.build/code-signing allIdentities=
  1) 33C5A99173A572E577DAF879FF82DDAE9DE2EFAE "iPhone Developer: m.schmidt@wind-energie.de (EAT66RHC99)"
  2) 3ACA6E5506DCB23306928AB0CFFAEB59C8568E86 "member: 04B3ADD5-C4FA-4D94-86D8-19619B71E8B2 28306F34-2F0B-406A-A2BF-B390AB7466DC" (CSSMERR_TP_NOT_TRUSTED)
  3) 3ACE07ED89D1AB5E182891B125D5D3E2B9D46B6F "Developer ID Installer: Johannes Me (XXXXXXXXX)"
  4) 33C4C262D5818C82FE57D9D8757327380CC3BB90 "Developer ID Application: Johannes Me (XXXXXXXXX)"
  5) 3DC0CFA50D30912D2A2C61A73AAE31052FB8533D "3rd Party Mac Developer Application: Johannes Me (XXXXXXXXX)"
  6) 34C786EC92AE42D97071E296CCB1A36D3DC7E645 "3rd Party Mac Developer Installer: Johannes Me (XXXXXXXXX)"
  7) 39C9AE8406A568E8B8CB19908D3C7BFE087F0838 "Apple Distribution: Johannes Me (XXXXXXXXX)"
     7 identities found

  Valid identities only
  1) 33C5A99173A572E577DAF879FF82DDAE9DE2EFAE "iPhone Developer: xxx@xxx (XXXXXXXX)"
  2) 3ACE07ED89D1AB5E182891B125D5D3E2B9D46B6F "Developer ID Installer: Johannes Me (XXXXXXXXX)"
  3) 33C4C262D5818C82FE57D9D8757327380CC3BB90 "Developer ID Application: Johannes Me (XXXXXXXXX)"
  4) 3DC0CFA50D30912D2A2C61A73AAE31052FB8533D "3rd Party Mac Developer Application: Johannes Me (XXXXXXXXX)"
  5) 34C786EC92AE42D97071E296CCB1A36D3DC7E645 "3rd Party Mac Developer Installer: Johannes Me (XXXXXXXXX)"
  6) 39C9AE8406A568E8B8CB19908D3C7BFE087F0838 "Apple Distribution: Johannes Me (XXXXXXXXX)"
     6 valid identities found
```

**Comment by ZhangWei-KUMO** [comment]
I have same problem，Did you resolve it?

**Comment by johannesjo** [comment]
@ZhangWei-KUMO yes. But I don't remember how exactly, sorry. I think in the end I started from scratch and then followed  the steps from the electron builder tutorial and the guides provided by apple very very strictly.

I am also using an export of the certificates and `CSC_LINK=exported-certs.p12`and `CSC_KEY_PASSWORD` in my build command.

You can check the rest of the setup here: https://github.com/johannesjo/super-productivity
But that's probably not very helpful.

**Comment by johannesjo** [comment]
@ZhangWei-KUMO I just run into the same problem again. Turns out I never properly installed the certificate for Mac Developer I only imported the provisioning profile... You have to go to: https://developer.apple.com/account/resources/certificates/ create the certifcate, download and then install it.

**Comment by semanticdiff** [comment]
Just chiming in here to add that I had the same problem. I resolved it by doing the same thing as @johannesjo. The electron builder documents suggest that we only need to generate the AppleDeveloper and AppleDistribution certificates. However, `electron-builder` (v23.0.3) seems to look for the `Mac Developer` certificate too on Keychain. So I went to https://developer.apple.com/account/resources/certificates/, created the certificate, downloaded it and then installed it. Electron-builder was able to then go to the next step.
